client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -315,8 +315,8 @@ const PatternAssembler = ( {
 		selectedCategory?: string | null
 	) => {
 		if ( selectedPattern ) {
-			// Inject the selected pattern category or the first category
-			// because it's used in tracks and as pattern name in the list
+			// Inject the selected pattern category (except for the category All)
+			// or the first category because it's used in tracks and as pattern name in the list
 			const [ firstCategory ] = Object.keys( selectedPattern.categories );
 			selectedPattern.category = categories.find(
 				( { name } ) => name === ( selectedCategory || firstCategory )
```

Comment:
I guess we can simply check the `selectedCategory` here and then we don't need to find the real category later

```js
selectedPattern.category = categories.find(
  ( { name } ) => {
    if ( selectedCategory ===  CATEGORY_ALL_SLUG ) {
      return name === firstCategory;
    }

    return name === ( selectedCategory || firstCategory );
  }
)
```